confmap/confmaptest/configtest_test.go

```diff
@@ -33,8 +33,7 @@ func TestLoadConf(t *testing.T) {
 func TestToStringMapSanitizeEmptySlice(t *testing.T) {
 	cfg, err := LoadConf(filepath.Join("testdata", "empty-slice.yaml"))
 	require.NoError(t, err)
-	var nilSlice []interface{}
-	assert.Equal(t, map[string]any{"slice": nilSlice}, cfg.ToStringMap())
+	assert.Equal(t, map[string]any{"slice": []interface{}{}}, cfg.ToStringMap())
```

Comment:
@yurishkuro What do you think of the change to this expectation here? It looks like it was conflicting with rest of the decoding flow. This is what `empty-slice.yaml` looks like:
```yaml
slice: [] # empty slices are sanitized to nil in ToStringMap
```
Given that `slice` is explicitly identified, shouldn't `slice` be empty instead of nil?

Reply:
if there is indeed a contradiction, I would suggest opening a ticket with description, to let the maintainers opine what behavior they expect.